mii/deployment.py

```diff
@@ -2,94 +2,70 @@
-
-        deployment_type: One of the ``enum mii.DeploymentTypes: [LOCAL]``.
-            *``LOCAL`` uses a grpc server to create a local deployment, and query the model must be done by creating a query handle using
-              `mii.mii_query_handle` and posting queries using ``mii_request_handle.query`` API,
-
-        model_path: Optional: In LOCAL deployments this is the local path where model checkpoints are available. In AML deployments this
-            is an optional relative path with AZURE_MODEL_DIR for the deployment.
-
-        enable_deepspeed: Optional: Defaults to True. Use this flag to enable or disable DeepSpeed-Inference optimizations
-
-        enable_zero: Optional: Defaults to False. Use this flag to enable or disable DeepSpeed-ZeRO inference
-
-        ds_config: Optional: Defaults to None. Use this to specify the DeepSpeed configuration when enabling DeepSpeed-ZeRO inference
-
-        force_register_model: Optional: Defaults to False. For AML deployments, set it to True if you want to re-register your model
-            with the same ``aml_model_tags`` using checkpoints from ``model_path``.
-
-        mii_config: Optional: Dictionary specifying optimization and deployment configurations that should override defaults in ``mii.config.MIIConfig``.
-            mii_config is future looking to support extensions in optimization strategies supported by DeepSpeed Inference as we extend mii.
-            As of now, it can be used to set tensor-slicing degree using 'tensor_parallel' and port number for deployment using 'port_number'.
-
-        version: Optional: Version to be set for AML deployment, useful if you want to deploy the same model with different settings.
-    Returns:
-        If deployment_type is `LOCAL`, returns just the name of the deployment that can be used to create a query handle using `mii.mii_query_handle(deployment_name)`
-
-    """
-
-    # parse and validate mii config
+from .config import MIIConfig, DeploymentType
+
+
+def support_legacy_api(
+    task,
+    model,
+    deployment_type=DeploymentType.LOCAL,
+    model_path="",
+    enable_deepspeed=True,
+    enable_zero=False,
+    ds_config=None,
+    mii_config=None,
+    version=1,
+):
+    if ds_config is None:
+        ds_config = {}
+    if mii_config is None:
+        mii_config = {}
+
+    deployment_config = {
+        "task": task,
+        "model": model,
+        "model_path": model_path,
+        "enable_deepspeed": enable_deepspeed,
+        "enable_zero": enable_zero,
+        "ds_config": ds_config,
+    }
+    for key, val in mii_config.items():
+        if not hasattr(MIIConfig, key):
+            deployment_config[key] = val
+
+    mii_config = {k: v for k, v in mii_config.items() if hasattr(MIIConfig, k)}
+    mii_config["version"] = version
+    mii_config["deployment_type"] = deployment_type
+
+    return deployment_config, mii_config
+
+
+def deploy(deployment_name: str,
```

Comment:
Would it be better to have a different API? Trying to think of the multi-model deployment case when designing the inputs. Imagine a multimodel setup:
```python
mii_config = {"port_number": 50000}
deployment_configs = [{"task":"text-generation", "model":"gpt2"}, {"task":"text-generation", "model":"bigscience/bloom-560m"}]
deployment_name = "my-multi-model-dep"
mii.deploy(deployment_name, deploymetn_configs, mii_config)
```

Can we find something better?